deploy/docker/start.sh

```diff
@@ -1,53 +1,53 @@
 #!/bin/sh
-set -e
+set -e -u
 
 emqx_exit(){
-    # tail erlang.log.*
-    erlang_log=$(echo $(ls -t /opt/emqx/log/erlang.log.*) | awk '{print $1}')
-    num=$(sed -n -e '/LOGGING STARTED/=' ${erlang_log} | tail -1)
-    tail -n +$((num-2)) ${erlang_log}
+    # At least erlang.log.1 exists
+    if [ -f /opt/emqx/log/erlang.log.1 ]; then
+        # tail emqx.log.*
+        erlang_log=$(echo $(ls -t /opt/emqx/log/erlang.log.*) | awk '{print $1}')
+        num=$(sed -n -e '/LOGGING STARTED/=' ${erlang_log} | tail -1)
+        [ ! -z $num ] && [ $num -gt 2 ] && tail -n +$(expr $num - 2) ${erlang_log}
+    fi
 
     echo "['$(date -u +"%Y-%m-%dT%H:%M:%SZ")']:emqx exit abnormally"
     exit 1
 }
 
 ## EMQ Main script
 
-# Start and run emqx, and when emqx crashed, this container will stop
+# When receiving the EXIT signal, execute emqx_exit function
+trap "emqx_exit" EXIT
 
-/opt/emqx/bin/emqx start || emqx_exit
+# Set log.rotation=off in etc/emqx.conf
+sed -i -r 's/log.rotation = (.*)/log.rotation = off/1' /opt/emqx/etc/emqx.conf
 
-tail -f /opt/emqx/log/erlang.log.1 &
+# Start and run emqx, and when emqx crashed, this container will stop
+/opt/emqx/bin/emqx start
 
 # Sleep 5 seconds to wait for the loaded plugins catch up.
 sleep 5
 
 echo "['$(date -u +"%Y-%m-%dT%H:%M:%SZ")']:emqx start"
 
+tail -f /opt/emqx/log/emqx.log &
+
 # monitor emqx is running, or the docker must stop to let docker PaaS know
 # warning: never use infinite loops such as `` while true; do sleep 1000; done`` here
 #          you must let user know emqx crashed and stop this container,
 #          and docker dispatching system can known and restart this container.
-NEW_LOG_NUM=2
 IDLE_TIME=0
 MGMT_CONF='/opt/emqx/etc/plugins/emqx_management.conf'
 MGMT_PORT=$(sed -n -r '/^management.listener.http[ \t]=[ \t].*$/p' $MGMT_CONF | sed -r 's/^management.listener.http = (.*)$/\1/g')
-while [[ $IDLE_TIME -lt 5 ]]; do
-    IDLE_TIME=$((IDLE_TIME+1))
+while [ $IDLE_TIME -lt 5 ]; do
+    IDLE_TIME=$(expr $IDLE_TIME + 1)
     if curl http://localhost:${MGMT_PORT}/status >/dev/null 2>&1; then
         IDLE_TIME=0
-        # Print the latest erlang.log
-        if [[ -f /opt/emqx/log/erlang.log.${NEW_LOG_NUM} ]];then
-            tail -f /opt/emqx/log/erlang.log.${NEW_LOG_NUM} &
-            [[ ! -z $(ps -ef |grep "tail -f /opt/emqx/log/erlang.log" | grep -vE "grep|erlang.log.${NEW_LOG_NUM}" | awk '{print $1}') ]] && kill $(ps -ef |grep "tail -f /opt/emqx/log/erlang.log" | grep -vE "grep|erlang.log.${NEW_LOG_NUM}" | awk '{print $1}') 
-            NEW_LOG_NUM=$((NEW_LOG_NUM+1))
-        fi
     else
         echo "['$(date -u +"%Y-%m-%dT%H:%M:%SZ")']:emqx not running, waiting for recovery in $((25-IDLE_TIME*5)) seconds"
     fi
     sleep 5
 done
 # If running to here (the result 5 times not is running, thus in 25s emqx is not running), exit docker image
 # Then the high level PaaS, e.g. docker swarm mode, will know and alert, rebanlance this service
-
-emqx_exit
+exit 1
```

Comment:
nit: there is no need to `exit` here, and the exit code is overridden by what `emqx_exit` exits with.